mod_wasm/modules/wasm/mod_wasm.c

```diff
@@ -304,26 +313,37 @@ static int content_handler(request_rec *r)
        * than what is needed, map all responses to a 500 error.
        */
 
-      if (ret != OK && ret != HTTP_OK) {
+      if (ret != OK && ret != HTTP_OK)
+      {
+        if (buffer != 0)
+            ap_log_rerror(APLOG_MARK, APLOG_ERR, ret, r, "ERROR! Invalid script response. On header #%i, found character '%s': %s", termarg, termch, buffer);
+
         if (r->content_type == NULL)
             ap_log_rerror(APLOG_MARK, APLOG_ERR, ret, r,
-                "content_handler() - ERROR! In WasmEnableCGI mode, HTTP headers are expected (i.e.: \"Content-type: text/html\n\n\")");
+                "ERROR! Couldn't identify mandatory 'Content-type' HTTP header (i.e.: \"Content-type: text/html\n\n\")");
 
-        wasm_return_const_char_ownership(module_response);
         return HTTP_INTERNAL_SERVER_ERROR;
       }
       if (termch != NULL) {
-        ap_rprintf(r, "%s", termch);
+        /*
+         * After parsing the response headers in the Wasm module output with
+         * #ap_scan_script_header_err_strs, the `termch` variable points to the
+         * last parsed character.
+         *
+         * To return the response body, we start from that character and calculate
+         * the remaining length in the response buffer.
+         */
```

Comment:
instead of this comment I would just use
```
int header_len = termch - module_response;
int body_len = header_len;
ap_rwrite(termch, body_len, r);
```

Reply:
I understand.
To avoid adding more local variables, I will improve the comments with a more straightforward step-by-step reasoning about how `headers_len` is calculated.